include/deal.II/hp/dof_level.h

```diff
@@ -296,15 +317,16 @@ namespace internal
       Assert (obj_index < dof_offsets.size(),
               ExcIndexRange (obj_index, 0, dof_offsets.size()));
 
-      // make sure we are on an
-      // object for which DoFs have
-      // been allocated at all
+      // make sure we are on an object for which DoFs have been
+      // allocated at all
       Assert (dof_offsets[obj_index] != (offset_type)(-1),
               ExcMessage ("You are trying to access degree of freedom "
                           "information for an object on which no such "
                           "information is available"));
 
-      Assert (fe_index == active_fe_indices[obj_index],
+      Assert (fe_index == ((signed_active_fe_index_type)active_fe_indices[obj_index]>=0 ?
+                           active_fe_indices[obj_index] :
+                           (active_fe_index_type)~(signed_active_fe_index_type)active_fe_indices[obj_index]),
```

Comment:
Why do you this? If I understand correctly, you cast to a `signed`, flip all the bits, and then recast to a `unsigned`.